programmers/난이도별/level02.H-Index/SSSOy.java

```diff
@@ -0,0 +1,22 @@
+import java.util.*;
+
+class Solution {
+    public int solution(int[] citations) {
+        int answer = 0;
+        Arrays.sort(citations);
+        
+        int index = citations[citations.length - 1];
+        while(true) {
+            answer = index--;
+            int cnt = 0;
+            
+            for(int i = 0; i < citations.length; i++) 
+                cnt += answer <= citations[i] ? 1 : 0;
+                
+            if(answer <= cnt || index < 0) 
+                break;
+        }
+        
+        return answer;
```

Comment:
```suggestion
        int answer = 0;
        int len = citations.length;
        Arrays.sort(citations);
        for (int i = 0; i < len; i++) {
            int h = len - i;
            answer = Math.max(answer, Math.min(citations[i], h));
        }
        return answer;
```
2중 반복문을 돌리지 않아도 풀이할 수 있습니다!

혹은 
```java
        int max = Arrays.stream(citations).reduce(0, Math::max);
        for (int i = max; i > -1; i--) {
            int h = i;
            if (Arrays.stream(citations).filter(v -> v >= h).count() >= i) return i;
        }
        return 0;
```
이런식으로 stream을 이용하여 풀이할수도 있어요~

Reply:
우와 이렇게 하면 더 간단하게 풀이할 수 있군요!!! 알려주셔서 감사합니다~